internal/namespaces/registry/v1/custom_docker_helper.go

```diff
@@ -0,0 +1,255 @@
+package registry
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"strings"
+
+	"github.com/scaleway/scaleway-cli/internal/core"
+	"github.com/scaleway/scaleway-cli/internal/interactive"
+	"github.com/scaleway/scaleway-cli/internal/printer"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+)
+
+const endpointPrefix = "rg."
+const endpointSuffix = ".scw.cloud"
+
+type emptyRequest struct{}
+
+type registryDockerHelperConfigureRequest struct {
+	HelperDirectory string
+}
+
+func registryDockerHelperConfigureCommand() *core.Command {
+	return &core.Command{
+		Short: `Configure a local Docker credential helper`,
+		Long: `This command will configure the Docker credential helper for your account.
+
+It will create a new script named docker-credential-scw. 
+This script will be called each time Docker needs the credentials and will return the correct credentials.
+It avoid running docker login commands.
+`,
+		Namespace: "registry",
+		Resource:  "configure-docker",
+		ArgsType:  reflect.TypeOf(registryDockerHelperConfigureRequest{}),
+		ArgSpecs: []*core.ArgSpec{
+			{
+				Name:  "helper-directory",
+				Short: "Directory in which the Docker helper will be installed",
+			},
+		},
+		Run: registryDockerHelperConfigureRun,
+	}
+}
+
+func registryDockerHelperConfigureRun(ctx context.Context, argsI interface{}) (i interface{}, e error) {
+	// TODO add windows support
+	if runtime.GOOS == "windows" {
+		return nil, fmt.Errorf("windows is not currently supported")
+	}
+
+	binaryName := core.ExtractBinaryName(ctx)
+
+	_, _ = interactive.Println("To enable the Docker credential helper, scw needs to create a script inside of your $PATH.")
+	scriptDirArg := argsI.(*registryDockerHelperConfigureRequest).HelperDirectory
+	if scriptDirArg == "" {
+		defaultScriptDirArg := filepath.Join(filepath.Dir(scw.GetConfigPath()), "bin")
+
+		promptedScriptDir, err := interactive.PromptStringWithConfig(&interactive.PromptStringConfig{
+			Prompt:          "In which directory do you want to install the script?",
+			DefaultValue:    defaultScriptDirArg,
+			DefaultValueDoc: defaultScriptDirArg,
+		})
+		if err != nil {
+			return nil, err
+		}
+		scriptDirArg = promptedScriptDir
+	}
+	profileFlag := ""
+	profileName := core.ExtractProfileName(ctx)
+	if profileName != "" {
```

Comment:
I would rather use an env var and add a condition to keep it if already exist
```
PROFILE_FLAG="-p test"
if SCW_PROFILE
  PROFILE_FLAG="-p $SCW_PROFIEL
```

Reply:
And change the precedence used by the cli ? :thinking: